plugin-energy-estimation-tdp/README.md

```diff
@@ -0,0 +1,75 @@
+
+- [introduction](#Introduction)
+- [Energy estimation tdp plugin](#Energy-estimation-tdp-plugin)
+  - [How to use](#how-to-use)
+  - [Prepare your environment](#prepare-your-environment)
+  - [Configuration](#Configuration)
+
+## Introduction
+
+This plugin estimate the energy consumption for pods based on the TDP value of the machine where pods are running.
+The TDP (Thermal Design Power) is the maximum amount of heat generated by a computer compnent. 
+Refer to https://en.wikipedia.org/wiki/Thermal_design_power for more details.
+
+In the first version of the plugin, we consider only the TDP of the CPU.
+
+ This plugin requires the cgroupv2 input plugin (k8s) as it needs the input measurements of cgroupsv2.
+
+ The estimation calculation is done using the following formula:
+
+ energy estimation = cgroupv2_cpu_total_usage * nb_vcpu/nb_cpu * tdp / 1000 000 / polling_interval
+
+cgroupv2_cpu_total_usage:   total usage of CPU in micro seconds for a pod 
+nb_vcpu:                    number of virtual CPU of the hosting machine where pod is running
+nb_cpu:                     number of physical CPU of the hosting machine where pod is running
+polling_interval:           polling interval of cgroupv2 input plugin
+
+
+## Energy estimation tdp plugin
+
+### How to use
+
+Just compile the app-agent of the alumet's github repository.
+
+```bash
+cargo run
+```
+
+
+The binary created by the compilation will be found under the target repository.
+
+### Prepare your environment
+
+To work this plugin needs k8s plugin configured, so the needed things are related to k8s plugin requirements:
+
+1. cgroup v2
+2. kubectl
+3. alumet-reader user
+
+### Configuration 
+
+[plugins.EnergyEstimationTdpPlugin]
+enable = true
+poll_interval = "30s"
+tdp = 100.0
+nb_vcpu = 1.0
+nb_cpu = 1.0
+
+pool_interval: must be identical to the poll_interval of input k8s plugin. Default value is 
+
+nb_vcpu: number of virtual cpu allocated to the virtual machine in case of kubernetes nodes are virtual machine. Using the kubectl get node command, you can retrieve the number of vcpu. If the kubernetes nodes are physical machine, assign it to value 1. Default value is 1.
+
+local@edge-master:~/manu/git/alumet$ kubectl  describe node edge-master.seed | grep cpu -B 2
+  Hostname:    edge-master.seed
+Capacity:
+  cpu:                32
+
+nb_cpu: number of physical cpu of the hosted machine. Using the lscpu or hwinfo, you can retrieve the number of cpu. If the kubernetes nodes are physical machine, assign it to value 1. Default value is 1.
+
+tdp: Thermal Design power; each CPU has a calculated thermal design value; the value can be find on internet (usually on CPU manufacturer); you need the exact CPU family (using command lscpu or hwinfo). For example, for Intel® Xeon® D Processor, family D-2183IT, the tpd can be found:
+https://ark.intel.com/content/www/us/en/ark/products/136441/intel-xeon-d-2183it-processor-22m-cache-2-20-ghz.html
+tdp value is  100W.
+Default value is 100.
```

Comment:
```suggestion
Default value is 100.

```

Reply:
I think he wants to add a trailing newline (standard)

Reply:
Yes exactly